UI/src/model/locationSearch.js

```diff
@@ -29,92 +29,95 @@
     };
 
     /**
-     * Combined numerical value search (asset id and road number, which is part of road address)
+     * Search by asset Id depending on current layer
      *
      * @param input
      * @returns {*}
      */
-    var idOrRoadNumber = function(input) {
+    var searchById = function(input) {
       if (selectedLayer === 'massTransitStop') {
-        return roadNumberAndNationalIdSearch(input);
+        return massTransitStopSearchByNationalId(input);
       } else if (selectedLayer === 'linkProperty') {
-        return roadNumberAndRoadLinkSearch(input);
+        return roadLinkSearchById(input);
       } else if (selectedLayer === 'speedLimit') {
-        return roadNumberAndSpeedLimitSearch(input.text);
+        return speedLimitSearchById(input);
       } else {
-        return getCoordinatesFromRoadAddress({roadNumber: input.text});
+        return $.Deferred().reject('Haulla ei löytynyt tuloksia');
       }
     };
 
     /**
-     * Speed limit id search, combined with road number search
+     * Speed limit id search
      *
      * @param input
      * @returns {*}
      */
-    var roadNumberAndSpeedLimitSearch = function(input){
-      var roadNumberSearch = backend.getCoordinatesFromRoadAddress(input);
-      var speedlimitSearch= backend.getSpeedLimitsLinkIDFromSegmentID(input);
-      return $.when(
-        speedlimitSearch, roadNumberSearch).then(function(speedlimitdata,roadData) {
-        var returnObject = roadLocationAPIResultParser(roadData);
-        if (_.get(speedlimitdata[0], 'success')) {
-          var linkid = _.get(speedlimitdata[0], 'linkId');
-          var y = _.get(speedlimitdata[0], 'latitude');
-          var x= _.get(speedlimitdata[0], 'longitude');
-          var title = input + " (nopeusrajoituksen ID)";
+    var speedLimitSearchById = function(input){
+      return $.when(backend.getSpeedLimitsLinkIDFromSegmentID(input.text)).then(function(speedlimitdata) {
+        var returnObject = [];
+        if (_.get(speedlimitdata, 'success')) {
```

Comment:
Comparing to previous,  speedlimitdata is an array.
Shouldn't we validate if it is an array?

Reply:
Don´t know why but it was never a array but a object